curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/65422ba173a18b1bedef1bb6.md

```diff
@@ -0,0 +1,658 @@
+---
+id: 65422ba173a18b1bedef1bb6
+title: Step 28
+challengeType: 0
+dashedName: step-28
+---
+
+# --description--
+
+Within the arrow function of the event listener, add an `if` statement to check if there is a currently playing song.
+
+The `if` statement should check if the `userData.currentSong` is `null`. If it is, then call the `playSong()` function with the `id` of the first song in the `userData.songs` array. 
+
+If it's not `null`, then proceed to the `else` statement.
+
+# --hints--
+
+Test 1
+
+```js
+
+```
```

Comment:
The tests for this step:

You should create an if statement with the condition `userData?.currentSong === null`.

```js
assert.match(code, /if\s*\(userData\?\.currentSong\s*===\s*null\)\s*\{/)
```

You should call the `playSong` function with `userData?.songs[0].id` inside your `if` block.

```js
assert.match(code, /if\s*\(userData\?\.currentSong\s*===\s*null\)\s*\{\s*playSong\(userData\?.songs\[0\]\.id\);?\s*\}/)
```